pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeFunctions.java

```diff
@@ -273,9 +274,23 @@ public static String toDateTime(long millis, String pattern, String timezoneId)
    */
   @ScalarFunction
   public static long fromDateTime(String dateTimeString, String pattern) {
+    if (StringUtils.isEmpty(dateTimeString)) {
```

Comment:
Any specific reason why we want to support empty string? IMO it should throw exception if the time value doesn't comply with the pattern

Reply:
Because of Tableau. Suppose a file as date time strings, but some of those values are NULL. since pinot doesn't support NULL, we store them as empty strings. 
When someone applies a date transform function on such a column, an error is thrown. 

However, the behavious should be a default value returned in those cases (or an explicit NULL value)

Reply:
I think it is still needed. e.g. https://github.com/apache/pinot/blob/master/pinot-tools/src/main/resources/examples/batch/githubEvents/githubEvents_offline_complexTypeHandling_table_config.json
In this what happens to `created_at_timestamp`, when `created_at` is null/empty ?

Reply:
I think it will throw NPE when `created_at` is `null`. My point is that we probably should not handle this in this general time handling function. The caller should catch the exception and fill the proper default value. The default value in the schema is not available to this function.

Reply:
removed now.